explainer.md

```diff
@@ -331,24 +331,53 @@ customElements.define("custom-tablist", CustomTabList);
+| `dismiss` or `escape` | _all elements_                                                    |                       | `Escape` KeyboardEvent             |
+| `contextMenu`         | _all elements_                                                    |                       | `contextmenu`                      |
+| `increment`           | Elements w/ computed role `progressbar`, `scrollbar`, or `slider` | vertical              | `Up` KeyboardEvent                 |
+|                       | ""                                                                | horizontal LTR        | `Right` KeyboardEvent              |
+|                       | ""                                                                | horizontal RTL        | `Left` KeyboardEvent               |
+|                       | Elements w/ computed role `spinbutton`                            | orientation n/a       | `Up` KeyboardEvent                 |
+| `decrement`           | Elements w/ computed role `progressbar`, `scrollbar`, or `slider` | vertical              | `Down` KeyboardEvent               |
+|                       | ""                                                                | horizontal LTR        | `Left` KeyboardEvent               |
+|                       | ""                                                                | horizontal RTL        | `Right` KeyboardEvent              |
+|                       | Elements w/ computed role `spinbutton`                            | orientation n/a       | `Down` KeyboardEvent               |
+| `scrollByPage`        | TBD (possibly custom scroll views)                                |                       | TBD (possibly `PageUp`/`PageDown`) |
+| `scrollIntoView`      | TBD                                                               |                       | No equivalent DOM event            |
+| `setValue`            | n/a                                                               |                       | No equivalent DOM event            |
+
+Notes on the previous table:
+- KeyboardEvent sequences include keyup/keydown/keypress.
+- Control orientation is determined by the computed value of `aria-orientation` which 
+  defaults to `horizontal` for `progressbar` and `slider`, and defaults to `vertical` for 
+  `scrollbar`.
+- Natural language direction is determined by the computed value of `dir` which defaults 
+  to `ltr`, but can be set to `rtl` for languages like Arabic and Hebrew.
+- The event target for KeyboardEvent sequences is the currently focused element.
+- The event target for `click` and `contextmenu` events is lowest leaf node DOM element in 
+  the assistive technology's "cursor" or "point-of-regard" .
+- If a web author does not cancel the event with `Event.preventDefault()` and/or 
+  `Event.stopPropagation()`, the event should propagate out of the web view an potentially 
```

Comment:
Hm, so this is basically saying that the code which synthesizes the keyboard event should check whether `preventDefault()` or `stopPropagation()` were called, and do some default handling if they weren't?

How does this factor in for things like `<input type="range">`? That has a default handling for an `increment` action - should it fire the keyboard events first? Or does this comment only apply to the events which don't have specific targets?

Reply:
> IIRC the default behavior of that control prevents the key events from propagating higher in the DOM tree.

That doesn't answer my question, unfortunately. Should we be firing keyboard events at `<input type="range">`, to be consistent, or skip the keyboard events entirely and directly set the value?

Reply:
Yikes I didn't intend that to be an edit of the original comment... Sorry for the faux pas.

> Should we be firing keyboard events at <input type="range">, to be consistent, or skip the keyboard events entirely and directly set the value?

TBD. I can see at least one potential impact, which may or may not be a downside:

- sending arrow on native range would only change value by the step value of the range, where as the value change is usually a percentage at the moment. I think the shift is different enough that users would notice.

…but I don't know if that or other potential downsides outweigh any potential benefit.